lib/Dialect/Torch/Transforms/ShapeLibrary.cpp

```diff
@@ -3497,60 +3497,65 @@ module {
   func.func @__torch__.torch.jit._shape_functions.conv_forwards(%arg0: !torch.list<int>, %arg1: !torch.list<int>, %arg2: !torch.optional<list<int>>, %arg3: !torch.list<int>, %arg4: !torch.list<int>, %arg5: !torch.list<int>, %arg6: !torch.bool, %arg7: !torch.list<int>, %arg8: !torch.int) -> !torch.list<int> {
     %true = torch.constant.bool true
     %int0 = torch.constant.int 0
-    %int2 = torch.constant.int 2
     %int1 = torch.constant.int 1
+    %int2 = torch.constant.int 2
     %0 = torch.aten.len.t %arg5 : !torch.list<int> -> !torch.int
     %1 = torch.aten.gt.int %0, %int0 : !torch.int, !torch.int -> !torch.bool
     %2 = torch.aten.len.t %arg0 : !torch.list<int> -> !torch.int
     %3 = torch.prim.ListConstruct  : () -> !torch.list<int>
-    %4 = torch.aten.__getitem__.t %arg0, %int0 : !torch.list<int>, !torch.int -> !torch.int
-    %5 = torch.aten.append.t %3, %4 : !torch.list<int>, !torch.int -> !torch.list<int>
-    %6 = torch.aten.__getitem__.t %arg1, %int0 : !torch.list<int>, !torch.int -> !torch.int
-    %7 = torch.aten.append.t %3, %6 : !torch.list<int>, !torch.int -> !torch.list<int>
-    %8 = torch.aten.__range_length %int2, %2, %int1 : !torch.int, !torch.int, !torch.int -> !torch.int
-    torch.prim.Loop %8, %true, init() {
+    %4 = torch.prim.If %arg6 -> (!torch.int) {
+      torch.prim.If.yield %int1 : !torch.int
+    } else {
+      torch.prim.If.yield %int0 : !torch.int
+    }
+    %5 = torch.aten.__getitem__.t %arg0, %int0 : !torch.list<int>, !torch.int -> !torch.int
```

Comment:
why make this change?

Reply:
This is auto changed by running update_shape_lib.sh

Reply:
Can you rebase to top of main? These changes were already added to torch-mlir in another PR: https://github.com/llvm/torch-mlir/pull/1286